lib/xcodeproj/project/object/helpers/file_references_factory.rb

```diff
@@ -126,7 +126,7 @@ def new_file_reference(group, path, source_tree)
           # @return [XCVersionGroup] The new reference.
           #
           def new_xcdatamodeld(group, path, source_tree)
-            path = Pathname.new(path)
+            path = Pathname.new(group.parent.path).realpath().join(group.path).join(path)
```

Comment:
Based on your suggestions though I did update the code as follows:
```
            if group.parent.path
              path = Pathname.new(group.parent.path).realpath.join(group.path).join(path)
            else
              path = Pathname.new(path)
            end
```
With this in place the tests pass again and datamodels in sub-directories still work